### PR TITLE
Cast rhel_version to string at the settings validation time

### PR DESCRIPTION
**Pull request by ogajduse**
Cast settings.server.version.rhel_version to string to explicitly state that string is the expected type.

I have also fixed `get_sat_rhel_version` function. `Satellite`'s `os_version` property returns Version object, so it fails if the function tries to do `Version(Version('1.1'))`

`6.11.z` cherry-pick would need an extra commit with the following patch:
```diff

diff --git a/tests/foreman/destructive/test_clone.py b/tests/foreman/destructive/test_clone.py
index 329dbb3b3..b6183404d 100644
--- a/tests/foreman/destructive/test_clone.py
+++ b/tests/foreman/destructive/test_clone.py
@@ -22,12 +22,13 @@ from robottelo import constants
 from robottelo.config import settings
 from robottelo.hosts import Satellite
 
+from robottelo.hosts import get_sat_rhel_version
 SSH_PASS = settings.server.ssh_password
 pytestmark = pytest.mark.destructive
 
 
 @pytest.mark.parametrize(
-    "sat_ready_rhel", [7, 8] if settings.server.version.rhel_version < 8 else [8], indirect=True
+    "sat_ready_rhel", [7, 8] if get_sat_rhel_version().major < 8 else [8], indirect=True
 )
 @pytest.mark.parametrize('backup_type', ['online', 'offline'])
 @pytest.mark.parametrize('skip_pulp', [False, True], ids=['include_pulp', 'skip_pulp'])
 ```

### CODE REVIEW

**Review by rplevka — APPROVED** [review]

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: -n 5 tests/foreman/api/test_ping.py

**Comment by jyejare** [comment]
Why this PR is in draft state @ogajduse ?

File: robottelo/config/validators.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
         Validator('server.hostnames', must_exist=True, is_type_of=list),
         Validator('server.version.release', must_exist=True),
         Validator('server.version.source', must_exist=True),
+        Validator('server.version.rhel_version', cast=int),
```
Comment (jyejare) [inline robottelo/config/validators.py:16]: What if we needed a floating version of RHEL somewhere in the tests tomorrow?

e.g :
```
if server.version.rhel_version == 8.7:
    <do this>
else:
    <do that>
```

Comment (jyejare) [reply robottelo/config/validators.py:16]: I mean there could be things we might had to handle based on RHEL minor version as well. Doint int casting will remove the minor version from here!

**Review by jyejare — COMMENTED** [review]
ACK pending question/concern!

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: -n 5 tests/foreman/api/test_ping.py

**Comment by omkarkhatavkar** [comment]
trigger: test-robottelo
pytest: -n 5 tests/foreman/api/test_ping.py

**Comment by ogajduse** [comment]
I am reopening the PR, all reviews were re-requested as the patch changed.

**Review by rplevka — APPROVED** [review]

**Pull request merged**